app/domain/authentication/resource_restrictions/validate_resource_restrictions.rb

```diff
@@ -66,10 +66,11 @@ def validate_resource_restrictions_configuration
       def validate_request_matches_resource_restrictions
         @logger.debug(LogMessages::Authentication::ResourceRestrictions::ValidatingResourceRestrictionsValues.new)
 
-        resource_restrictions.each do |restriction_name, restriction_value|
-          if @authentication_request.retrieve_attribute(restriction_name) != restriction_value
-            raise Errors::Authentication::ResourceRestrictions::InvalidResourceRestrictions, restriction_name
-          end
+        resource_restrictions.each do |name, value|
```

Comment:
The `name, value` here looks like a [data clump](https://refactoring.guru/smells/data-clumps) to me, representing a `Restriction` object.  Like one of my other comments, fixing this _fully_ is an existing code problem and probably out of scope, since it would also affect the code that saves annotations to the db (I didn't investigate that, but it may be a rabbit hole). 

I leave it to you to decide if you want to introduce it now for this code.

Reply:
Thanks for this proposal, @jonahx.
I think combining the `name` and `value` into one object can solve this data clump you mentioned correctly.
Regarding your second proposal, as much as it sounds promising, it requires a complex code to decide which restriction to create.
Since the same code is called for all authenticators (k8s, azure, gcp) this requires a new class as input, to decide which restriction to create.
I see where this is going, and it looks to me less concise and clear.

I did the first part, though!